### DISCUSSION TITLE
Se hace necesaria una etapa SoapRequest

### DISCUSSION THREAD

**Discussion by seiturra** [Cotalker/documentation#371] [3. Ideas & Feature Requests] [answered]
Buenas a todos,

En Operaciones se ha presentado con varios clientes que utilizan SAP la necesidad de crear código personalizado para generar las llamadas de red hacia sus servidores cuando utilizan SOAP como protocolo de comunicación. 

Sería bueno si se pudiese generar una etapa de bots/survey triggers/etc que permita generar estas llamadas, similar a la actual NWRequest pero en versión SOAP.

Gracias y buena semana.

**Comment by acedward** [comment]
¿Cómo podría funcionar con el WSDL?

**Reply by seiturra** [reply]
Ya que se necesita tener un cliente con todas las definiciones de WSDL, podrían ser dos etapas que se usen en conjunto:
etapa1: Crear cliente. Input: URL del WSDL.
etapa2: Utilizar las funciones disponibles por la etapa anterior. Input dependerá del output de la etapa anterior. Sería como NWR, con un campo sin estructura que sólo recibe un objeto (o arreglo?) que luego se transforma en el input del WS.

Posible mejora para etapa 1: aceptar archivo de texto o url para crear el cliente.

**Comment by guillevalin** [comment, marked as answer]
@SeIturra Puedes usar Network Request para hacer este tipo de llamadas también

Método: POST
URL: Endpoint SOAP al cual deseas llamar.
Headers: Debes pasar "Content-Type: text/xml; charset=UTF-8"
Authorization: Pasar el Basic Auth o token dependiendo de lo que necesites.
Body: Aquí va el XML tal cual.

Esto no funcionaría con servicios SOAP que usan funcionalidades como WS Security, pero para la mayoría de los endpoints con Basic Auth podrías usarlo.

**Reply by seiturra** [reply]
Gracias @guillevalin! 

Serán desafíos las autenticaciones que requieran estar en base 64 o generar el xml, ya que las bibliotecas hacen la conversión de objeto a xml por lo general. Es lo que recuerdo que me trajo algunos dolores cuando los trabajé las primeras veces. De todos modos, para el próximo desarrollo con SOAP lo podremos intentar si este feature request se califica como no necesario.
